bindings/profiler.cc

```diff
@@ -101,37 +101,158 @@ CpuProfiler* cpuProfiler = CpuProfiler::New(v8::Isolate::GetCurrent());
 CpuProfiler* cpuProfiler = v8::Isolate::GetCurrent()->GetCpuProfiler();
 #endif
 
+Local<Object> CreateTimeNode(Local<String> name, Local<String> scriptName, 
+                         Local<Integer> scriptId, Local<Integer> lineNumber,
+                         Local<Integer> columnNumber, Local<Integer> hitCount,
+                         Local<Array> children) {
+  Local<Object> js_node = Nan::New<Object>();
+  js_node->Set(Nan::New<String>("name").ToLocalChecked(), name);
+  js_node->Set(Nan::New<String>("scriptName").ToLocalChecked(), scriptName);
+  js_node->Set(Nan::New<String>("scriptId").ToLocalChecked(), scriptId);
+  js_node->Set(Nan::New<String>("lineNumber").ToLocalChecked(), lineNumber);
+  js_node->Set(Nan::New<String>("columnNumber").ToLocalChecked(), columnNumber);
+  js_node->Set(Nan::New<String>("hitCount").ToLocalChecked(), hitCount);
+  js_node->Set(Nan::New<String>("children").ToLocalChecked(), children);
+  return js_node;
+}
+
+
+#if NODE_MODULE_VERSION > NODE_11_0_MODULE_VERSION
+Local<Object> TranslateLineNumbersTimeProfileNode(const CpuProfileNode* parent,
+                                      const CpuProfileNode* node);
+
+Local<Array> GetLineNumberTimeProfileChildren(const CpuProfileNode* parent,
+                                      const CpuProfileNode* node) {
+  unsigned int index = 0;
+  Local<Array> children;
+  int32_t count = node->GetChildrenCount();
+
+  unsigned int hitLineCount = node->GetHitLineCount();
+  unsigned int hitCount = node->GetHitCount();
+  if (hitLineCount > 0) {
+    std::vector<CpuProfileNode::LineTick> entries(hitLineCount);
+    node->GetLineTicks(&entries[0], hitLineCount);
```

Comment:
I'm not a C++ stylist, so I am not sure if this generally accepted style: is it acceptable to pass the vector internals when a C-style array is expected? My concern is that this code is depending on the implicit assumption that a resize will not happen.

Reply:
Feel free to consult with a C++ stylist on this one as react appropriately.

Reply:
I didn't need a vector for any reason. Switched to array.